### PR TITLE
feat: add a REST sample and test for AML AI

### PR DESCRIPTION
**Pull request by irataxy**
## Description

https://cloud.google.com/financial-services/anti-money-laundering/docs/concepts/overview

Fixes b:355523386

Note: Before submitting a pull request, please open an issue for discussion if you are not associated with Google.

## Checklist

- [X] I have followed [Sample Format Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [X] `pom.xml` parent set to latest `shared-configuration`
- [ ] Appropriate changes to README are included in PR
- [X] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] **Tests** pass:   `mvn clean verify` **required**
- [X] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [ ] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [X] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [X] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample 
- [X] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: aml-ai/src/main/java/com/example/amlai/ListLocations.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+public class ListLocations {
+
+  public static void main(String[] args) throws Exception {
+    // TODO(developer): Replace these variables before running the sample.
+    String projectId = "my-project-id";
+
+    listLocations(projectId);
+  }
+
+  public static String listLocations(String projectId) throws IOException {
```
[the remark this answers is not in the document]
Comment (irataxy) [reply aml-ai/src/main/java/com/example/amlai/ListLocations.java:40]: Done

File: aml-ai/src/test/java/com/example/amlai/ListLocationsTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  @BeforeClass
+  public static void checkRequirements() {
+    requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
+    PROJECT_ID = requireEnvVar("GOOGLE_CLOUD_PROJECT");
+  }
+
+  @Test
+  public void test_ListLocations() throws IOException {
+    String output = ListLocations.listLocations(PROJECT_ID);
+    assertThat(output, containsString(LOCATION));
```
[the remark this answers is not in the document]
Comment (irataxy) [reply aml-ai/src/test/java/com/example/amlai/ListLocationsTest.java]: Done

File: aml-ai/src/test/java/com/example/amlai/ListLocationsTest.java
[the remark this answers is not in the document]
Comment (irataxy) [reply aml-ai/src/test/java/com/example/amlai/ListLocationsTest.java:1]: Done

**Comment by Sita04** [comment]
@irataxy Thanks for addressing the changes! PTAL at the build errors.

**Pull request closed without merging**